Add python binding for `generate_debug_handles`

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57481 [Pytorch Delegated Backend] Save function name in debug info
* #57441 [Pytorch Delegated Backend] Save operator name and function name in
* **#57156 [Pytorch Delegated Backend] Add python binding for**
* #55462 [Pytorch Backend delegation] Add api for backend lowering to query debug

generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

kimishpatel committed:
[Pytorch Delegated Backend] Add python binding for
generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for generate_debug_handles"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

raziel commented on torch/csrc/jit/backends/backend_init.cpp:
```diff
@@ -179,6 +180,12 @@ void initJitBackendBindings(PyObject* module) {
             "Object ", py::str(orig_module), " is not a ScriptModule"));
       });
+
+  m.def(
+      "_jit_backend_generate_debug_handles",
+      [](std::shared_ptr<Graph>& graph) {
+         return generate_debug_handles(graph);
+      });
```
This is going to change per the follow up discussed in PR/55462

So that PR should be done first.

kimishpatel replied:
@raziel so my plan was to land all the diffs in this stack first and then do the PR that changes preprocess signature. That is a public API with few places where it might be used. And since it touches many sites, it may take longer to land. I did not want that to block progress on what we want to deliver this half.

raziel reviewed (approved):
Sounds good. Let's then get this in.
Thanks

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).

kimishpatel committed:
Update on "[Pytorch Delegated Backend] Add python binding for"


generate_debug_handles

Summary:
To be able to generate debug handles for preprocess written inpython.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28062328](https://our.internmc.facebook.com/intern/diff/D28062328)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28062328).
